app/src/main/java/com/keylesspalace/tusky/ComposeActivity.java

```diff
@@ -1429,6 +1436,26 @@ public void onEmojiSelected(@NotNull String shortcode) {
         textEditor.getText().insert(textEditor.getSelectionStart(), ":"+shortcode+": ");
     }
 
+    private void loadCachedInstanceMetadata(@NotNull AccountEntity activeAccount)
+    {
+        InstanceEntity instanceEntity = TuskyApplication.getDB().instanceDao().loadMetadataForInstance(activeAccount.getDomain());
+
+        if(instanceEntity != null) {
+            Integer max = instanceEntity.getMaximumTootCharacters();
+            maximumTootCharacters = (max == null ? STATUS_CHARACTER_LIMIT : max.intValue());
+            emojiList = instanceEntity.getEmojiList();
+            if (emojiList != null) {
+                emojiView.setAdapter(new EmojiAdapter(emojiList, ComposeActivity.this));
+            }
+        }
+    }
+
+    private void cacheInstanceMetadata(@NotNull AccountEntity activeAccount)
```

Comment:
I wonder why it works.
You call it from requests callbacks which are called... on the main thread I guess?
Shouldn't Room complain that you access DB from UI thread?

Reply:
We set `allowMainThreadQueries()` in `TuskyApplication`

Reply:
Uuuugh, afaik it's mostly for debugging... I wish we did it asynchronously but okay

Reply:
Yes its ugly. We will get rid of it eventually.